src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1372,6 +1372,66 @@ void C2_MacroAssembler::minmax_fp(FloatRegister dst, FloatRegister src1, FloatRe
   bind(Done);
 }
 
+// According to Java SE specification, for floating-point round operations, if
+// the input is NaN, +/-infinity, or +/-0, the same input is returned as the
+// rounded result; this differs from behavior of RISC-V fcvt instructions (which
+// round out-of-range values to the nearest max or min value), therefore special
+// handling is needed by NaN, +/-Infinity, +/-0.
+void C2_MacroAssembler::round_double_mode(FloatRegister dst, FloatRegister src, int round_mode, Register tmp1, Register tmp2, Register tmp3) {
+
+  assert_different_registers(dst, src);
+  assert_different_registers(tmp1, tmp2, tmp3);
```

Comment:
Suggestion: `assert_different_registers(dst, src, tmp1, tmp2, tmp3);`

Reply:
I doubt we can use assert_different_registers with different typos of register classes ( Register and FloatRegister here)

Reply:
That's right. Thanks for pointing this out.